src/common/utils/channel/request-channel-injection-token.ts

```diff
@@ -4,17 +4,19 @@
  */
 
 import { getInjectionToken } from "@ogre-tools/injectable";
-import type { JsonValue } from "type-fest";
+import type { IpcValue } from "./allowed-types";
 
 export interface RequestChannel<
-  Request extends JsonValue | void = void,
-  Response extends JsonValue | void = void,
+  Request extends IpcValue | void = void,
+  Response extends IpcValue | void = void,
 > {
-  id: string;
-  _requestSignature?: Request;
-  _responseSignature?: Response;
+  id: Request | Response extends boolean ? string : string;
```

Comment:
Shouldn't `id` be always a string? I don't understand what is being attempted here.

Reply:
It is, but I realise that I should have just added some ignore lint lines instead. Will add some descriptive comments.